beacon_node/beacon_chain/src/metrics.rs

```diff
@@ -1009,6 +1009,16 @@ lazy_static! {
         "beacon_aggregated_attestation_subsets_total",
         "Count of new aggregated attestations that are subsets of already known aggregates"
     );
+    pub static ref VALIDATOR_MONITOR_MISSED_NON_FINALIZED_BLOCKS_AT_NON_FINALIZED_EPOCH_TOTAL: Result<IntGaugeVec> = try_create_int_gauge_vec(
+        "validator_monitor_missed_non_finalized_blocks_at_non_finalized_epoch_total",
+        "Number of non-finalized blocks missed at non-finalized epochs",
+        &["src", "validator"]
+    );
+    pub static ref VALIDATOR_MONITOR_MISSED_NON_FINALIZED_BLOCKS_TOTAL: Result<IntGaugeVec> = try_create_int_gauge_vec(
```

Comment:
Since this is an always increasing value, the ideal Prometheus data type is an `int_counter`.

Reply:
I decided to going for a gauge as a counter can only be incremented and not set to a specific value.

In the case where we'd want to change to a counter, we'd need to prune the entire Hashmap once the Prometheus metrics are updated and increment the missed block metric everytime there's an entry in the Hashmap.

Reply:
Oh yep, makes sense! Sorry, I must have been confused!